packages/store/src/decorators/select/select-factory.ts

```diff
@@ -1,20 +1,57 @@
 import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
 
 import { Store } from '../../store';
-import { NgxsConfig } from '../../symbols';
+import { META_KEY, NgxsConfig } from '../../symbols';
+import { propGetter } from '../../internal/internals';
+import { CONFIG_MESSAGES, VALIDATION_CODE } from '../../configs/messages.config';
 
 /**
  * Allows the select decorator to get access to the DI store.
- * @internal
+ * @internal only use in @Select decorator
  * @ignore
  */
 @Injectable()
 export class SelectFactory {
   public static store: Store | null = null;
   public static config: NgxsConfig | null = null;
+  private static readonly DOLLAR_CHAR_CODE: number = 36;
 
   constructor(store: Store, config: NgxsConfig) {
     SelectFactory.store = store;
     SelectFactory.config = config;
   }
+
+  /**
+   * @example If `foo$` => make it just `foo`
+   */
+  public static removeDollarAtTheEnd(name: string): string {
```

Comment:
This doesn't improve code readability. I'm not a fan of this "Java" approach with classes and static members, and it definitely shouldn't be in our library as we're developing in TypeScript! Functions are first-class citizens in JavaScript.

There should be a clear purpose of moving `removeDollarAtTheEnd` from the `function removeDollarAtTheEnd` way to the `static removeDollarAtTheEnd`. What's the benefit?

`SelectFactory` is a service locator that should contain references to the `Store` and `NgxsConfig` instances, but it shouldn't contain internal API as static members.

Please, revert it back :pray:

Reply:
I’ll think about how this can be improved, I hate functions in JS, and all my projects are saturated with classes, since I am a backend developer in Kotlin and Java.

Reply:
@splincode 

Dude, you're very good at development, but let's stick with the JS approach as we're developing in JS :heart_decoration:

Reply:
I understand, I will fix it

Reply:
done